backends/p4tools/modules/testgen/targets/bmv2/test/BMV2MetadataXfail.cmake

```diff
@@ -69,6 +69,12 @@ p4tools_add_xfail_reason(
   bmv2_assert.p4
 )
 
+p4tools_add_xfail_reason(
+  "testgen-p4c-bmv2-metadata"
+  "Unsupported type argument for Value Set"
```

Comment:
What happend to these tests? Are they broken by this PR somehow, or were they not really working before too and just now started actually failing? Is this something part 3 will resolve?

Reply:
`p4c-bm2-ss` throws the same error when invoked on the program. It means that this P4 program does not produce a valid P4Runtime API:
```
build/p4c-bm2-ss --target bmv2 --arch v1model --p4runtime-files out.txt testdata/p4_16_samples/pvs-nested-struct.p4
testdata/p4_16_samples/pvs-nested-struct.p4(22): [--Werror=unsupported] error: Unsupported type argument for Value Set; this version of P4Runtime requires that when the type parameter of a Value Set is a struct, all the fields of the struct must be of type bit<W>, but inner is not
  inner_value_set_t inner;
  ```
We could avoid the error by not invoking the P4RuntimeSerializer for certain test back ends, but you will still run into a similar problem when trying to run the program. The Xfail already existed for the PTF tests, for instance.